File: src/main/java/com/swp391/koibe/repositories/BidHistoryRepository.java
package com.swp391.koibe.repositories;

import com.swp391.koibe.models.Bid;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface BidHistoryRepository extends JpaRepository<Bid, Long> {

    Boolean existsByAuctionKoiIdAndBidderId(Long auctionKoiId, Long bidderId);
    List<Bid> getAllByAuctionKoiId(Long auctionKoiId);

}

File: src/main/java/com/swp391/koibe/repositories/BidHistoryRepository.kt
package com.swp391.koibe.repositories

import com.swp391.koibe.models.Bid
import org.springframework.data.jpa.repository.JpaRepository

interface BidHistoryRepository : JpaRepository<Bid, Long> {
    fun existsByAuctionKoiIdAndBidderId(auctionKoiId: Long, bidderId: Long): Boolean
    fun getAllByAuctionKoiId(auctionKoiId: Long): ArrayList<Bid>
}
